controllers/topology/current_state.go

```diff
@@ -117,6 +117,13 @@ func (r *ClusterReconciler) getCurrentControlPlaneState(ctx context.Context, clu
 	if err != nil {
 		return nil, errors.Wrapf(err, "failed to get InfrastructureMachineTemplate for %s", tlog.KObj{Obj: res.Object})
 	}
+	// FIXME(TBD) should we do this? If yes, we should do the same for the MD templates
```

Comment:
TBD: I saw that we check for the label on the InfraCluster and ControlPlane. Do we want to do this for the other templates too?

Reply:
I don't have objections, but let's move to a separate issue given that this is not a nit?

Reply:
Good point, removed from this PR

Reply:
> https://github.com/kubernetes-sigs/cluster-api/issues/5760